cmd/generator/add.go

```diff
@@ -30,11 +28,22 @@ var addCmd = &cobra.Command{
 	RunE: func(command *cobra.Command, args []string) error {
 		command.SilenceUsage = true
 
-		plugins, err := InitCommand(command)
+		err := InitCommand(command)
 		if err != nil {
 			return err
 		}
 
+		db, err := openDatabase(command)
+		if err != nil {
+			return errors.Wrap(err, "failed to open plugin database")
+		}
+		defer db.Close()
+
+		plugins, err := model.PluginsFromReader(db)
```

Comment:
One different approach I've considered was using the Store (https://github.com/mattermost/mattermost-marketplace/blob/master/internal/store/plugin.go)  instead. But that would require some changes to the Store and the interface was not as clean as I liked it to be.

Reply:
I'm fine with deferring the store changes, but instead of passing around an `os.File` that requires rewinding, should we just have helper methods akin to `pluginsFromDatabase` and `pluginsToDatabase`?

Reply:
Good suggestion :+1: